stablehlo/reference/Ops.cpp

```diff
@@ -541,6 +546,23 @@ Tensor evalLogisticOp(const Tensor &operand, TensorType resultType) {
   return result;
 }
 
+Tensor evalMapOp(ArrayRef<Tensor> inputs, Region &computation, Scope &scope,
+                 TensorType resultType) {
+  Tensor result(resultType);
+  for (auto inputIt = inputs[0].index_begin(); inputIt != inputs[0].index_end();
```

Comment:
IMO this top level iterator is used only to get the result index. In that case, let it be a iterator on result.